group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupMetadataManager.java

```diff
@@ -3962,7 +4359,167 @@ public void replay(
             }
             removeGroup(groupId);
         }
+    }
+
+    /**
+     * Replays StreamsGroupMemberMetadataKey/Value to update the hard state of
+     * the streams group.
+     * It updates the subscription part of the member or deletes the member.
+     *
+     * @param key   A StreamsGroupMemberMetadataKey key.
+     * @param value A StreamsGroupMemberMetadataValue record.
+     */
+    public void replay(
+        StreamsGroupMemberMetadataKey key,
+        StreamsGroupMemberMetadataValue value
+    ) {
+        String groupId = key.groupId();
+        String memberId = key.memberId();
+
+        StreamsGroup streamsGroup;
+        try {
+            streamsGroup = getOrMaybeCreatePersistedStreamsGroup(groupId, value != null);
+        } catch (GroupIdNotFoundException ex) {
+            // If the group does not exist and a tombstone is replayed, we can ignore it.
+            return;
+        }
+
+        if (value != null) {
+            StreamsGroupMember oldMember = streamsGroup.getOrMaybeCreateMember(memberId, true);
+            streamsGroup.updateMember(new StreamsGroupMember.Builder(oldMember)
+                .updateWith(value)
+                .build());
```

Comment:
I find it confusing that the Streams group object creates a member without adding it to its members and the member needs to be added with `updateMember()`. I know that it is done this way for the other groups. Creating a member outside of the group should not be a concern of the group. 

You do not need to change anything. Just a thought that I had for discussion.

Reply:
I understand. I was not advertizing to add the member to the group within `getOrMaybeCreate()`, but rather to explicitly build the member outside of `getOrMaybeCreate()`. We would then have a simple `get()` method and depeding on the result of the get use the `Builder` of the member. That would make it quite clear that the member is not added to the group. But maybe a renaming of `getOrMaybeCreate()` to `getOrDefault()` or similar would also make it clear.